lib/code_corps/task/service.ex

```diff
@@ -86,10 +86,9 @@ defmodule CodeCorps.Task.Service do
   defp update_on_github(%Task{github_repo_id: _, github_issue_id: nil} = task), do: task |> create_on_github()
   defp update_on_github(%Task{github_repo: github_repo} = task) do
     with {:ok, payload} <- GitHub.API.Issue.update(task),
-         {:ok, %GithubIssue{}} <-
-           payload
-           |> Sync.Issue.GithubIssue.create_or_update_issue(github_repo) do
-      {:ok, Task |> Repo.get(task.id)}
+         {:ok, %GithubIssue{}} <- payload |> Sync.Issue.GithubIssue.create_or_update_issue(github_repo),
+         task <- Task |> Repo.get(task.id) do
```

Comment:
If this is nil, what happens?

Reply:
Could it be, since the main function expects the `task`?

Reply:
I could move it back out of the `with`, though.

Reply:
It's low likelihood but imaging an update and a delete happening in parallel. Could add a `when not(is_nil(task))` guard and handle the nil case in the else.